### PR TITLE
feat(CORSO-53): add a service

### PR DESCRIPTION
**Pull request by Mila5847**
JIRA: link to jira ticket
Ticket: CORSO-53
Link: https://champlainsaintlambert.atlassian.net/browse/CORSO-53
## Context:
The ticket is about allowing an admin to add a new service to the services that CORSO offers.
## Changes
- Creating the "View All Services Through Admin Dashboard" page.
- Creating the "Add Service" page.
- Saving the images/icons of newly added services to Cloudinary.
- Testing the "add service" functionality.
- Testing the front-end (E2E).
## Before
No UI
### After
image
image
## Dev notes (Optional)
- Specific technical changes that should be noted
## Linked pull requests (Optional)
- pull request link

### CODE REVIEW

**Review by KarinaSofia — APPROVED** [review]
Good work! looks nice and its well tested!

File: corso_website_springboot/src/main/java/com/corso/springboot/Service_Subdomain/businesslayer/CloudinaryService.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+        cloudinary = new Cloudinary(ObjectUtils.asMap(
+                "cloud_name", "dszhbawv7",
+                "api_key", "533582356614965",
+                "api_secret", "katPxshTkWq5wdvRa24ukrjqrLI",
+                "secure", true));
+    }
+
+    public String uploadBase64Image(String base64Image) throws IOException {
+        if (base64Image == null || base64Image.isEmpty()) {
+            throw new IllegalArgumentException("Cannot upload because the base64Image is null or empty.");
```
Comment (DylanBrass) [inline corso_website_springboot/src/main/java/com/corso/springboot/Service_Subdomain/businesslayer/CloudinaryService.java:27]: This message should probably be more english, this is a bit too technical for a user...

File: corso_website_springboot/src/main/java/com/corso/springboot/Service_Subdomain/businesslayer/CloudinaryService.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+            // Decode the base64 string
+            byte[] decodedBytes = Base64.getDecoder().decode(base64ImageNoPrefix);
+
+            // Upload the decoded bytes and retrieve the URL
+            @SuppressWarnings("unchecked")
+            Map uploadResult = cloudinary.uploader().upload(decodedBytes, ObjectUtils.emptyMap());
+            String url = (String) uploadResult.get("url");
+
+            if (url == null || url.isEmpty()) {
+                throw new IllegalStateException("Failed to upload the image; no URL was returned from Cloudinary.");
```
Comment (DylanBrass) [inline corso_website_springboot/src/main/java/com/corso/springboot/Service_Subdomain/businesslayer/CloudinaryService.java:43]: Same thing for this, all the errors, maybe find the cause of this...

**Review by DylanBrass — APPROVED** [review]
Looks good but try to understand the errors better and the actuall process of cloudinary...

File: corso_website_springboot/src/main/java/com/corso/springboot/Service_Subdomain/businesslayer/CloudinaryService.java (hunk 1 above)
Comment (Mila5847) [reply corso_website_springboot/src/main/java/com/corso/springboot/Service_Subdomain/businesslayer/CloudinaryService.java:27]: Yep, you are right, did not think about this, i will look into it!

**Pull request merged**